api/v1alpha1/gateway_extensions_types.go

```diff
@@ -62,16 +62,35 @@ type GatewayExtensionSpec struct {
 	// +optional
 	RateLimit *RateLimitProvider `json:"rateLimit,omitempty"`
 
-	// JWTProviders configures named JWT providers.
+	// JWT configuration for JWT extension type.
+	// +optional
+	JWT *JWT `json:"jwt,omitempty"`
+}
+
+type JWT struct {
+	// ValidationMode configures how JWT validation behaves.
+	// If unset or empty, STRICT mode is used (JWT is required).
+	// If set to AllowMissing, requests without JWT tokens are allowed to pass through.
```

Comment:
nit, worth considering if making additional changes:

```suggestion
	// If set to AllowMissing, unauthenticated requests without a JWT are allowed through.
	// If using this mode, make sure you consider the security implications and 
	// consider using an `RBAC` policy to enforce authorization.
```